go/release_handler.go

```diff
@@ -0,0 +1,377 @@
+		tag:        event.GetRelease().GetTagName(),
+		draft:      event.GetRelease().GetDraft(),
+		prerelease: event.GetRelease().GetPrerelease(),
+		url:        event.GetRelease().GetHTMLURL(),
+	}
+}
+
+type ReleaseHandler struct {
+	githubapp.ClientCreator
+	botLogin string
+
+	m sync.Mutex
+}
+
+func NewReleaseHandler(cc githubapp.ClientCreator, botLogin string) (h *ReleaseHandler, err error) {
+	h = &ReleaseHandler{
+		ClientCreator: cc,
+		botLogin:      botLogin,
+	}
+	err = os.MkdirAll(h.Workdir(), 0777|os.ModeDir)
+
+	return h, err
+}
+
+func (h *ReleaseHandler) Workdir() string {
+	return filepath.Join("/", "tmp", "release_handler")
+}
+
+func (h *ReleaseHandler) Handles() []string {
+	return []string{"release"}
+}
+
+func (h *ReleaseHandler) Handle(ctx context.Context, _, _ string, payload []byte) error {
+	var event github.ReleaseEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return errors.Wrap(err, "Failed to parse release event payload")
+	}
+
+	switch event.GetAction() {
+	case "published":
+		releaseMeta := getReleaseMetadata(&event)
+		if releaseMeta.repoName != "vitess" {
+			return nil
+		}
+
+		if releaseMeta.draft {
+			return nil
+		}
+
+		version, err := semver.Parse(releaseMeta.tag)
+		if err != nil { // release tag is not semver-compliant (which includes release candidates)
+			return nil
+		}
+
+		client, err := h.NewInstallationClient(githubapp.GetInstallationIDFromEvent(&event))
+		if err != nil {
+			return err
+		}
+
+		h.m.Lock()
+		defer h.m.Unlock()
+
+		_, err = h.updateReleasedCobraDocs(ctx, client, releaseMeta, version)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// TODO: refactor out shared code between here and synchronizeCobraDocs()
+func (h *ReleaseHandler) updateReleasedCobraDocs(
+	ctx context.Context,
+	client *github.Client,
+	releaseMeta *releaseMetadata,
+	version semver.Version,
+) (*github.PullRequest, error) {
+	vitess := git.NewRepo(
+		releaseMeta.repoOwner,
+		"vitess",
+	).WithLocalDir(filepath.Join(h.Workdir(), "vitess"))
+	website := git.NewRepo(
+		releaseMeta.repoOwner,
+		"website",
+	).WithLocalDir(filepath.Join(h.Workdir(), "website"))
+
+	prs, err := website.ListPRs(ctx, client, github.PullRequestListOptions{
+		State:     "open",
+		Head:      "update-release-cobradocs-for-",
+		Base:      "prod",
+		Sort:      "created",
+		Direction: "desc",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	branch := "prod"
+	baseRepo := website
+	op := "update release cobradocs"
+
+	for _, pr := range prs {
+		if pr.GetUser().GetLogin() != h.botLogin {
+			continue
+		}
+
+		// Most recent PR created by the bot. Base a new PR off of it.
+		head := pr.GetHead()
+
+		branch = head.GetRef()
+		repo := head.GetRepo()
+		baseRepo = git.NewRepo(repo.GetOwner().GetLogin(), repo.GetName())
+
+		zerolog.DefaultContextLogger.Debug().Msgf("using existing PR #%d (%s/%s:%s)", pr.GetNumber(), baseRepo.Owner, baseRepo.Name, branch)
+		break
```

Comment:
I was thinking, the bot can create other PRs than the one that update the docs and version pairs. Shouldn't we also check that the most recent PR is a cobra doc update?